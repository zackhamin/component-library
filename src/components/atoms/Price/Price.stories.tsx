import { Text } from '../Text';
import { Price, PriceProps } from './Price';

export default {
	title: 'Atoms/Price',
	component: Price,
};

export const PriceStory = (args: PriceProps) => (
	<Text>
		<Price {...args} />
	</Text>
);

PriceStory.storyName = 'Price';
PriceStory.args = {
	amount: 9.99,
	currency: 'GBP',
	locale: 'en-GB',
};
PriceStory.argTypes = {
	currency: {
		options: ['GBP', 'EUR', 'USD'],
		control: { type: 'select' },
	},
	locale: {
		options: ['en-GB', 'en-IE', 'nl-NL', 'de-DE', 'fr-FR', 'nl-BE'],
		control: { type: 'select' },
	},
};
