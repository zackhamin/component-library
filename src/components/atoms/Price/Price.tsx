import { Locale, Price as PriceType } from '@Types';

export interface PriceProps extends PriceType {
	locale?: Locale;
}

export const Price = ({ currency, amount, locale = 'en-GB' }: PriceProps) => {
	const parsedAmount = typeof amount !== 'number' ? parseFloat(amount) : amount;

	return (
		<>
			{new Intl.NumberFormat(locale, {
				style: 'currency',
				currency,
				minimumFractionDigits: 2,
			}).format(parsedAmount)}
		</>
	);
};
