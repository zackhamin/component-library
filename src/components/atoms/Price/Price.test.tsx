import { Price } from './Price';
import { render } from '@testing-library/react';

describe('Price', () => {
	it('renders correctly', () => {
		const { container } = render(<Price amount={9.99} currency="GBP" />);
		expect(container.firstChild).toMatchSnapshot();
	});
	it('Should format an float string correctly', () => {
		const { container } = render(<Price amount={'9.99'} currency="GBP" />);

		expect(container.textContent).toBe('£9.99');
	});

	it('Should format an int string correctly', () => {
		const { container } = render(<Price amount={'9'} currency="GBP" />);

		expect(container.textContent).toBe('£9.00');
	});

	it('Should format a float number correctly', () => {
		const { container } = render(<Price amount={9.99} currency="GBP" />);

		expect(container.textContent).toBe('£9.99');
	});

	it('Should format an int number correctly', () => {
		const { container } = render(<Price amount={9} currency="GBP" />);

		expect(container.textContent).toBe('£9.00');
	});
});
