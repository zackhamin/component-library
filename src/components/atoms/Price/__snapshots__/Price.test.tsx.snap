// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Price renders correctly 1`] = `Â£9.99`;
